workers/unity/Packages/io.improbable.gdk.core/.codegen/Source/Generators/Core/UnityTypeGenerator.cs

```diff
@@ -0,0 +1,30 @@
+using Improbable.Gdk.CodeGeneration.CodeWriter;
+using Improbable.Gdk.CodeGeneration.Model.Details;
+using NLog;
+
+namespace Improbable.Gdk.CodeGenerator
+{
+    public static class UnityTypeGenerator
+    {
+        private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
+
+        public static string Generate(UnityTypeDetails details, string package)
+        {
+            var qualifiedNamespace = package;
+
+            return CodeWriter.Populate(cgw =>
+            {
+                cgw.UsingDirectives(
+                    "System.Linq",
+                    "Improbable.Gdk.Core",
+                    "UnityEngine"
+                );
+
+                cgw.Namespace(qualifiedNamespace, ns =>
+                {
+                    ns.Type(UnityTypeContent.Generate(details, qualifiedNamespace));
```

Comment:
Can this not be inlined now?

Reply:
worth coming back to on the post-tt-purge codegen pass 

not so easy because the generator has the `using` and `namespace`, whereas the content is just the type class/struct